workflow/controller/opeartor_data_test.go

```diff
@@ -0,0 +1,247 @@
+package controller
+
+import (
+	"context"
+	"fmt"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+
+	wfv1 "github.com/argoproj/argo-workflows/v3/pkg/apis/workflow/v1alpha1"
+)
+
+var inMemoryDataNode = `
```

Comment:
200 lines of code - anyway we can reduce the maintenance this might need by making it somehow more focused?

Reply:
It's 200 lines of node status, not code :) I have already reduced this as much as possible